diff -Nur -x '*.orig' -x '*~' zeroc-ice/cpp/src/IceUtil/Cond.cpp zeroc-ice.new/cpp/src/IceUtil/Cond.cpp
--- zeroc-ice/cpp/src/IceUtil/Cond.cpp	2008-05-16 19:24:01.000000000 +0200
+++ zeroc-ice.new/cpp/src/IceUtil/Cond.cpp	2008-08-25 16:04:06.368110208 +0200
@@ -307,7 +307,7 @@
         throw ThreadSyscallException(__FILE__, __LINE__, rc);
     }
 
-#if !defined(__hpux) && !defined(__APPLE__)
+#if !defined(__hppa) && !defined(__FreeBSD_kernel__) && !defined(__APPLE__)
     rc = pthread_condattr_setclock(&attr, CLOCK_MONOTONIC); 
     if(rc != 0)
     {
diff -Nur -x '*.orig' -x '*~' zeroc-ice/cpp/src/IceUtil/Time.cpp zeroc-ice.new/cpp/src/IceUtil/Time.cpp
--- zeroc-ice/cpp/src/IceUtil/Time.cpp	2008-05-16 19:24:01.000000000 +0200
+++ zeroc-ice.new/cpp/src/IceUtil/Time.cpp	2008-08-25 16:04:56.443331174 +0200
@@ -111,7 +111,7 @@
 #  endif
             return Time(static_cast<Int64>(tb.time) * ICE_INT64(1000000) + tb.millitm * 1000);
         }
-#elif defined(__hpux) || defined(__APPLE__)
+#elif defined(__hppa) || defined(__FreeBSD_kernel__) || defined(__APPLE__)
         //
         // HP/MacOS does not support CLOCK_MONOTONIC
         //
