diff -Nurp '--exclude=.git*' linux-2.6.35.9/drivers/media/video/uvc/uvc_driver.c beagle/drivers/media/video/uvc/uvc_driver.c
--- linux-2.6.35.9/drivers/media/video/uvc/uvc_driver.c	2011-05-09 11:23:46.175708955 +0200
+++ beagle/drivers/media/video/uvc/uvc_driver.c	2011-05-11 11:17:10.419677950 +0200
@@ -350,7 +350,7 @@ static int uvc_parse_format(struct uvc_d
 		strlcpy(format->name, "MJPEG", sizeof format->name);
 		format->fcc = V4L2_PIX_FMT_MJPEG;
 		format->flags = UVC_FMT_FLAG_COMPRESSED;
-		format->bpp = 0;
+		format->bpp = 4;
 		ftype = UVC_VS_FRAME_MJPEG;
 		break;
 
@@ -463,10 +463,16 @@ static int uvc_parse_format(struct uvc_d
 		 * uncompressed formats this can be fixed by computing the
 		 * value from the frame size.
 		 */
-		if (!(format->flags & UVC_FMT_FLAG_COMPRESSED))
+		if (!(format->flags & UVC_FMT_FLAG_COMPRESSED) ||
+				(format->type == UVC_VS_FORMAT_MJPEG))
 			frame->dwMaxVideoFrameBufferSize = format->bpp
 				* frame->wWidth * frame->wHeight / 8;
 
+		printk("---> framesize %ix%i, buffsize %i, bpp %i\n",
+			frame->wWidth, frame->wHeight,
+			frame->dwMaxVideoFrameBufferSize,
+			format->bpp);
+
 		/* Some bogus devices report dwMinFrameInterval equal to
 		 * dwMaxFrameInterval and have dwFrameIntervalStep set to
 		 * zero. Setting all null intervals to 1 fixes the problem and
diff -Nurp '--exclude=.git*' linux-2.6.35.9/drivers/media/video/uvc/uvc_video.c beagle/drivers/media/video/uvc/uvc_video.c
--- linux-2.6.35.9/drivers/media/video/uvc/uvc_video.c	2011-05-09 11:23:46.175708955 +0200
+++ beagle/drivers/media/video/uvc/uvc_video.c	2011-05-11 11:17:10.419677950 +0200
@@ -91,7 +91,8 @@ static void uvc_fixup_video_ctrl(struct
 		ctrl->dwMaxVideoFrameSize =
 			frame->dwMaxVideoFrameBufferSize;
 
-	if (!(format->flags & UVC_FMT_FLAG_COMPRESSED) &&
+	if ((!(format->flags & UVC_FMT_FLAG_COMPRESSED) ||
+	    (format->type == UVC_VS_FORMAT_MJPEG)) &&
 	    stream->dev->quirks & UVC_QUIRK_FIX_BANDWIDTH &&
 	    stream->intf->num_altsetting > 1) {
 		u32 interval;
